app/services/teacher_vacancy_authorisation.rb

```diff
@@ -23,13 +24,20 @@ def all_permissions
     end
 
     def school_urn
-      user_permissions.any? ? user_permissions.first['school_urn'] : nil
+      return nil unless user_permissions.any?
+      return user_permissions_for_school.first['school_urn'] if @school_urn.present?
+      user_permissions.first['school_urn']
     end
 
     def many?
       user_permissions && user_permissions.count > 1
     end
 
+    def authorised?
+      return user_permissions.any? if @school_urn.blank?
```

Comment:
I'm confident in our ability to associate our users during the invitation having looked at the first 60 so I think that if DfE Sign-in doesn't return an organisation that we can match on with a permission, we fail and return unauthorised. 

My previous comment described why I think it's risky to authorise a user in this scenario where a user doesn't have an organisation in DSI but has any other granted authorisation with us, we allow them in as the first organisation we find. I think this can safely go. Do you have any worries about losing it?

We may later look at displaying a friendly page in future that says "Hey we can see you're in this organisation but you've not been granted access by this organisation to use this service yet. Do x,y,z to make it happen".

Reply:
Roger, I guess I was just trying to break as few things as possible, but in hindsight it doesn't matter as we're reimporting everyone anyway. I'll make it so the organisation returned from DSI has to match one for the user's permission in TVA.